Support OTG via config.txt

ajpenner opened the pull request:
Support Ethernet over USB

This is my first contribution to a yocto main project, so if I am missing anything please do let me know I am happy to resubmit with any missing data.

**- What I did**
I added support for "on the go" via both config.txt and cmdline.txt
 
**- How I did it**
In both cases I use new configuration variables (ENABLE_DWC2_OTG, and CMDLINE_RNDIS) which is expected to be supplied in the main project's conf/local.conf. These modify the boot setup for the raspberry pi, in alignment with the recommended practise for the same setup when using Rasbian.

kraj commented on recipes-bsp/bootfiles/rpi-config_git.bb:
```diff
@@ -225,6 +225,12 @@ do_deploy() {
+    # DWC2 USB OTG support
+    if [ "${ENABLE_DWC2_OTG}" = "1" ] && [ "${ENABLE_DWC2_PERIPHERAL}" != "1" ]; then
+        echo "# Enable USB OTG mode" >> ${DEPLOYDIR}/bcm2835-bootfiles/config.txt
+        echo "dtoverlay=dwc2,dr_mode=otg" >> ${DEPLOYDIR}/bcm2835-bootfiles/config.txt
```
I think using `$CONFIG` instead of hardcoding the file name is better here

kraj commented on recipes-bsp/bootfiles/rpi-config_git.bb:
```diff
@@ -225,6 +225,12 @@ do_deploy() {
     fi
+    
+    # DWC2 USB OTG support
+    if [ "${ENABLE_DWC2_OTG}" = "1" ] && [ "${ENABLE_DWC2_PERIPHERAL}" != "1" ]; then
```
does this work with non-bash shells ?

ajpenner replied:
To the best of my knowledge (please correct me if I am wrong), but Yocto wants us to work with dash, where this logic does work. The syntax for if-else changes for csh or tcsh, so this scripting will not work there.

kraj replied:
Yes that should be enough

kraj commented on recipes-bsp/bootfiles/rpi-cmdline.bb:
```diff
@@ -29,6 +29,11 @@ CMDLINE_LOGO ?= '${@oe.utils.conditional("DISABLE_RPI_BOOT_LOGO", "1", "logo.nol
 CMDLINE_DEBUG ?= ""
 
+# Add RNDIS capabilities (must be after rootwait)
+# example: modules-load=dwc2,g_ether g_ether.host_addr=<some MAC address> g_ether.dev_addr=<some MAC address>
```
limit lines to 80 columns

agherzan commented:
Thanks for your contribution. Have a look at the documentation for how to format git log and other guidelines: https://meta-raspberrypi.readthedocs.io/en/latest/contributing.html. You have missed the signed-off-by line.

ajpenner commented:
> Thanks for your contribution. Have a look at the documentation for how to format git log and other guidelines: https://meta-raspberrypi.readthedocs.io/en/latest/contributing.html. You have missed the signed-off-by line.

I have rebased the changes with the requested changes (and added the Signed-off-by line).

agherzan commented:
NIT: ` bootfiles: Support ethernet over USB ` -> ` rpi-cmdline: Support ethernet over USB `

agherzan commented:
@agherzan-ci ok to test

ajpenner commented:
> NIT: `bootfiles: Support ethernet over USB` -> `rpi-cmdline: Support ethernet over USB`

@agherzan Is the above a change I should make?

agherzan commented:
@ajlennon it would be great if you could.

ajlennon commented:
Hiya :) Not me I think?

agherzan commented:
Unless you really insist :) (sorry for the ping). ajpenner ^

agherzan commented:
@ajpenner I've merged this as it is but it sounds like we need to update the docs too. Could you handle it as a separate PR?
